src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let newStr = '';
+
+    for (const i of this) {
+      if (i != null) {
+        newStr += i;
+      }
+
+      if (this[this.length - 1] !== i) {
+        newStr += separator;
+      }
```

Comment:
what happens when the last element of the array is repeated somewhere else in the array. 
for example `[1, 2, 4, 3, 4].join2()`
can you predict why it won't work correctly?

Reply:
it was necessary to compare the indices. while i compare items.
If the element is the same as the last one, the separator will not work

corrected